tools/cleanup_duplicate_host_guest_devices.rb

```diff
@@ -45,6 +46,20 @@
 
 return if opts[:dry_run]
 
-GuestDevice.destroy(guest_devices_to_delete) unless opts[:dry_run]
+dependents = GuestDevice.reflect_on_all_associations(:has_one) + GuestDevice.reflect_on_all_associations(:has_many)
+dependent_destroys = dependents.select { |assoc| assoc.options[:dependent] == :destroy }
```

Comment:
may want to check for `delete` too.

and if we handle `:delete`, maybe we can separate them and use `destroy` or `delete` accordingly

Reply:
Yeah thats a good call, maybe just find the dependent key and call the value method?

So if it is options[:dependent] => :delete would call .delete

This is starting to get closer to general enough to behave the way I expected .destroy_all to work